apps/meteor/server/services/messages/service.ts

```diff
@@ -0,0 +1,24 @@
+import { Users } from '@rocket.chat/models';
+import type { IMessage } from '@rocket.chat/core-typings';
+
+import { ServiceClassInternal } from '../../sdk/types/ServiceClass';
+import type { IMessageService } from '../../sdk/types/IMessageService';
+import { createDirectMessage } from '../../methods/createDirectMessage';
+import { executeSendMessage } from '../../../app/lib/server/methods/sendMessage';
+
+export class MessageService extends ServiceClassInternal implements IMessageService {
+	protected name = 'message';
+
+	async createDirectMessage({ to, from }: { to: string; from: string }): Promise<{ rid: string }> {
+		const [toUser, fromUser] = await Promise.all([Users.findOneByUsername(to), Users.findOneByUsername(from)]);
+
+		if (!toUser || !fromUser) {
+			throw new Error('error-invalid-user');
+		}
+		return createDirectMessage([to], from);
```

Comment:
@KevLehman Can you please re-check this? I feel there is some bug here. While testing I found that `to` and `from` params were both `userIds` and not `usernames`

Reply:
Ohhh, you're right. Since i use users with same id/username i didn't catch this. But indeed, it should be findById